### ISSUE TITLE
Sync with github account is requested every app launch

### ISSUE THREAD

**Issue by morallys** [microsoft/vscode#142799]
Nós escrevemos os dados necessários em sua área de transferência porque era muito grande para ser enviado. Por favor, Cole.

Issue Type: <b>Bug</b>

Since I formatted my linux, Manjaro, and installed the versions available in pamac, which are:

visual-studio-code-bin, snap and flatpak.

In all of them, synchronization with the github account, liveshare, etc. is requested at every restart of the application. I've already installed and reinstalled several times, including deleting the cache and files created by default, and still the synchronization is requested.

Is there anything I can do to fix this problem?

VS Code version: Code 1.64.2 (f80445acd5a3dadef24aa209168452a3d97cc326, 2022-02-09T22:02:29.527Z)
OS version: Linux x64 5.15.19-1-MANJARO snap
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 2337)|
|GPU Status|2d_canvas: unavailable_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: disabled_off<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|1, 1, 1|
|Memory (System)|7.68GB (1.14GB free)|
|Process Argv|--no-sandbox --force-user-env --unity-launch --crash-reporter-id 65b790ce-eed3-4f10-a419-539b1e7c13bf|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|plasma|
|XDG_CURRENT_DESKTOP|KDE|
|XDG_SESSION_DESKTOP|KDE|
|XDG_SESSION_TYPE|x11|
</details><details><summary>Extensions (19)</summary>

Extension|Author (truncated)|Version
---|---|---
jest-snippets|and|1.8.0
bracket-pair-colorizer|Coe|1.0.62
vscode-eslint|dba|2.2.2
es7-react-js-snippets|dsz|4.3.2
gitlens|eam|11.7.0
prettier-vscode|esb|9.2.0
comment-anchors|Exo|1.9.6
auto-close-tag|for|0.5.14
auto-rename-tag|for|0.1.10
code-runner|for|0.11.7
vscode-language-pack-pt-BR|MS-|1.64.6
vsliveshare|ms-|1.0.5330
material-icon-theme|PKi|4.13.0
LiveServer|rit|5.7.4
synthwave-vscode|Rob|0.1.11
react-proptypes-generate|sum|1.7.6
errorlens|use|3.4.1
quokka-vscode|Wal|1.0.445
material-theme|zhu|3.13.19

(5 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyl392:30425749
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
pythondataviewer:30285071
vscod805cf:30301675
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
vsc1dst:30433059
pythonvs932:30410667
wslgetstartedc:30433508
vsrem710:30416614
pythonvsnew555cf:30436485
vsbas813:30436447

```

</details>

![image](https://user-images.githubusercontent.com/44041037/153531946-e87f79c0-67c6-40be-b61d-80e74d9e19c7.png)

**Comment by TylerLeonhardt** [comment]
Can you follow these steps so that we can get more information on what's going on?

* Open the Command Palette
* Run the `Developer: Set Log Level...` command
* Choose `Trace`
* _Wait for the issue to happen..._
* Open the Output pane (ctrl+shift+u I think)
* drop in this issue he contents of the following:
  * `GitHub Authentication` Output channel
  * `Log (Settings Sync)` Output channel
  * `Log (Window)` Output channel

**Comment by Misery7100** [comment]
@TylerLeonhardt 
Nothing new as I can see. There is some time difference due to timezone settings, just ignore it.

**On sign in**

_1. GitHub Authenticatior_
```
[Info  - 19:13:17.296] Logging in for the following scopes: user:email
[Trace  - 19:13:22.827] Handling Uri...
[Info  - 19:13:22.827] Exchanging code for token...
[Info  - 19:13:23.472] Token exchange success!
[Info  - 19:13:23.473] Getting user info...
[Info  - 19:13:23.815] Got account info!
[Info  - 19:13:23.815] Storing 1 sessions...
[Info  - 19:13:23.824] Reading sessions from keychain...
[Info  - 19:13:23.824] Stored 1 sessions!
[Info  - 19:13:23.824] Login success!
[Trace  - 19:13:23.828] Token acquired from secret storage.
[Info  - 19:13:23.828] Got stored sessions!
[Trace  - 19:13:23.829] Read the following session from the keychain with the following scopes: user:email
[Info  - 19:13:23.829] Got 1 verified sessions.
[Info  - 19:13:23.830] Getting sessions for user:email...
[Info  - 19:13:23.830] Getting sessions for user:email...
[Info  - 19:13:23.831] Got 1 sessions for user:email...
[Info  - 19:13:23.831] Got 1 sessions for user:email...
```

_2. Log (Settings Sync)_
```
[2022-02-17 20:13:59.618] [settingssync] [info] Auto Sync: Triggered by Activity
[2022-02-17 20:13:59.752] [settingssync] [info] Settings: No changes found during synchronizing settings.
[2022-02-17 20:13:59.755] [settingssync] [info] Keybindings: No changes found during synchronizing keybindings.
[2022-02-17 20:13:59.757] [settingssync] [info] Snippets: No changes found during synchronizing snippets.
[2022-02-17 20:13:59.758] [settingssync] [info] Tasks: No changes found during synchronizing tasks.
[2022-02-17 20:13:59.760] [settingssync] [info] GlobalState: No changes found during synchronizing ui state.
[2022-02-17 20:13:59.792] [settingssync] [info] Extensions: No changes found during synchronizing extensions.
[2022-02-17 20:13:59.793] [settingssync] [info] Sync done. Took 76ms
```

_3. Log (Window)_
```
[2022-02-17 20:13:23.826] [renderer1] [info] Settings Sync: Updated current session 2100b5ae-e543-420d-85e6-9a34c70d01dc
```

**After restart**

_1. GitHub Authenticatior_
```
[Info  - 19:16:29.278] Reading sessions from keychain...
[Info  - 19:16:29.311] Getting sessions for user:email...
[Info  - 19:16:29.311] Getting sessions for all scopes...
[Info  - 19:16:29.328] Got 0 sessions for user:email...
[Info  - 19:16:29.328] Got 0 sessions for ...
```

_2. Log (Settings Sync)_
```
[2022-02-17 20:16:29.414] [settingssync] [info] Using settings sync service https://vscode-sync.trafficmanager.net/
[2022-02-17 20:16:29.414] [settingssync] [info] Auto Sync is enabled.
[2022-02-17 20:16:29.414] [settingssync] [info] Auto Sync: Suspended until auth token is available.
```

_3. Log (Window)_
```
[2022-02-17 20:16:28.051] [renderer1] [info] IExtensionHostStarter.start() took 26 ms.
[2022-02-17 20:16:28.184] [renderer1] [error] [Extension Host] (node:120757) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `code --trace-deprecation ...` to show where the warning was created)
```

**Comment by TylerLeonhardt** [comment]
do you have gnome-keyring installed on your system?

**Comment by TylerLeonhardt** [comment]
Hmm nothing in the logs pointed me to that. It was just a hunch. I wish there was a way for me to detect if `gnome-keyring` was the problem but I'm not sure there's anything I can do with this. Glad you are all unblocked.

**Comment by renaudcerrato** [comment]
After restart (logged out) :

1. GitHub Authentication:

```
[Info  - 11:42:58.496] Reading sessions from keychain...
[Info  - 11:42:58.523] Getting sessions for user:email...
[Info  - 11:42:58.548] Got 0 sessions for user:email...
[Info  - 11:42:58.608] Getting sessions for all scopes...
[Info  - 11:42:58.608] Got 0 sessions for ...
[Info  - 11:43:11.687] Getting sessions for all scopes...
[Info  - 11:43:11.687] Got 0 sessions for ...
```

2. Log Window
```
[2022-02-18 12:42:57.451] [renderer1] [info] IExtensionHostStarter.start() took 95 ms.
[2022-02-18 12:42:57.574] [renderer1] [error] [Extension Host] (node:15732) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `code --trace-deprecation ...` to show where the warning was created)
```

3. Log Main
```
[2022-02-18 12:42:56.507] [main] [info] update#setState idle
[2022-02-18 12:42:56.537] [main] [info] ExtensionHostStarterWorker created
[2022-02-18 12:42:57.367] [main] [info] Starting extension host with pid 15732 (fork() took 10 ms).
[2022-02-18 12:42:57.367] [main] [info] ExtensionHostStarterWorker.start() took 11 ms.
[2022-02-18 12:42:57.725] [main] [warning] Switching to using in-memory credential store instead because Keytar failed to load: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.secrets was not provided by any .service files
[2022-02-18 12:43:26.508] [main] [info] update#setState checking for updates
[2022-02-18 12:43:26.521] [main] [info] update#setState available for download
```

4. Settings Sync

```
[2022-02-18 12:42:57.684] [settingssync] [info] Using settings sync service https://vscode-sync.trafficmanager.net/
[2022-02-18 12:42:57.684] [settingssync] [info] Auto Sync is enabled.
[2022-02-18 12:42:57.684] [settingssync] [info] Auto Sync: Suspended until auth token is available.
```

**Comment by TylerLeonhardt** [comment]
@renaudcerrato secrets on Linux in vscode are stored using `org.freedesktop.secrets` which is suppose to be an abstraction layer over keyrings that implement it properly.

With that said, I've never had much success in understanding if KDE has a keyring/wallet that is compatible, even though my reading says that the primary use case was for abstracting over gnome and KDE... I can tell you for sure that `gnome-keyring` works, but I need some help from folks who use KDE to understand if KDE actually has an alternative or not and that status of that.

This link:
https://forum.authpass.app/t/what-is-the-provider-of-org-freedesktop-secrets-for-kde-plasma-5/162

Says:

> Here’s a current answer from a KDE developer, so I consider this to the be official answer. He said:
> 
> > KWallet still doesn’t implement org.freedesktop.secrets, and KSecretService is dead.

That's from August.

I have seen several folks mention that KeePassXC is apparently a solution (although they don't mention it in the context of vscode) so that might be a valid alternative for us.

In short, my recommendation is:
* provide feedback to KDE to finish their implementation of `org.freedesktop.secrets`
* use gnome-keyring or give KeePassXC a try and let me know how it goes!
